worker/buildbot_worker/test/unit/test_runprocess.py

```diff
@@ -857,3 +857,20 @@ def test_statFile_exists(self):
         self.assertEqual(
             st and st[2], 2, "statfile.log exists and size is correct")
         os.remove('statfile.log')
+
+    def test_invalid_utf8(self):
+        # create the log file watcher first
+        rp = self.makeRP()
+        lf = runprocess.LogFileWatcher(rp, 'test', 'invalid_utf8.log',
```

Comment:
Could we use `tempfile.NamedTemporaryFile`? If tests are ran in parallel and any other test uses this path we'll get rare failures that will be hard to track down.